tests/e2e/gateway_test.go

```diff
@@ -0,0 +1,113 @@
+package e2e_test
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/matchers/jq"
+
+	. "github.com/onsi/gomega"
+)
+
+const (
+	gatewayName          = "odh-gateway"
+	gatewayClassName     = "odh-gateway-class"
+	gatewayNamespace     = "openshift-ingress"
+	gatewayServiceName   = "default-gateway"
+	gatewayTLSSecretName = "default-gateway-tls"
+)
+
+var (
+	GatewayServiceGVK = schema.GroupVersionKind{
+		Group:   "services.platform.opendatahub.io",
+		Version: "v1alpha1",
+		Kind:    "GatewayConfig",
+	}
+
+	GatewayClassGVK = schema.GroupVersionKind{
+		Group:   "gateway.networking.k8s.io",
+		Version: "v1",
+		Kind:    "GatewayClass",
+	}
+
+	GatewayGVK = schema.GroupVersionKind{
+		Group:   "gateway.networking.k8s.io",
+		Version: "v1",
+		Kind:    "Gateway",
+	}
+
+	SecretGVK = schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Secret",
+	}
+)
```

Comment:
These should be moved to the `gvk` package, as I believe we will use them further.